src/main/java/apaintus/controllers/CanvasController.java

```diff
@@ -200,4 +200,20 @@ public void changed(ObservableValue<? extends Double> observableValue, Double ol
             }
         }
     }
+    
+    public class TextAreaChangeListener implements ChangeListener<String> {
+        private Attribute attribute;
+
+        public TextAreaChangeListener(Attribute attribute) {
+            this.attribute = attribute;
+        }
+
+        @Override
+        public void changed(ObservableValue<? extends String> observableValue, String oldValue, String newValue) {
+            if(activeShape != null && activeShape.isSelected() && drawnShapes.contains(activeShape)) {
```

Comment:
does this means that the text can only change when a shape is selected from inside the attribute bar ?

Reply:
I have not modified that part of the code, the only changes on that section is the spacing on the left